Assignment6.Rmd

```diff
@@ -70,14 +79,59 @@ table(M2$certified, M2$predict1)
 
 table(M2$certified, M2$predict2)
 
+#tree2 has lower error rate. because the sum of "no"+"no" and "yes"+"yes" are 5363 > 2186 in tree1. 
+
+#error rate in predict1
+M2$count1<- ifelse(M2$predict1 == M2$certified, 1, 0)
+M2$count2<- ifelse(M2$predict2 == M2$certified, 1, 0)
+
+mean(M2$count1) #0.2186
+mean(M2$count2) #0.5363
+
+#tree2 has higher prediction accuracy. 
+
 ```
 
+
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
 
+D1<- read.csv("student.course.csv")
+#because the original dataset is too big, so I randomly select 1000 of them. 
+D2<- D1[sample(nrow(D1), 1000), ]
+
+#Model 1 that include all the raw variables to predict GPAO. 
+c.tree3 <- rpart(GPAO ~ ANONID + CATALOG_NBR + SUBJECT + GRD_PTS_PER_UNIT + DIV + ANON_INSTR_ID + TERM, data = D2)
+
+printcp(c.tree3)
+
+rpart.plot(c.tree3)
+
+#Model 2: feature selection/extractrion
+D3<-dplyr::select(D2, GPAO, GRD_PTS_PER_UNIT, TERM, CATALOG_NBR, ANONID, ANON_INSTR_ID)
+D3<- sapply(D3, as.numeric )
+D4<- data.frame(scale(D3))
+
+#use pca 
+pca <- prcomp(D4, scale = TRUE)
+
+pca1<- data.frame(pca$x)
+
+D5<- dplyr::bind_cols(D2, pca1)
+
+#c.tree4<- rpart(GPAO ~ GRD_PTS_PER_UNIT + TERM , data= D2)
+c.tree5<- rpart(GPAO ~ PC1 + PC2, data = D5)
+printcp(c.tree5)
+
+rpart.plot(c.tree5)
+
+
+#Discussion: Both two models are used to predict GPAO. The first model use all raw variables in the data, creating a 8-split decision tree that has cross validation error of 0.728 and standard error of 0.127. The second model use PC1 and PC2 to predit GPAO, creating a 8-split decision tree that has cross validation error of 0.417 and standard error of 0.067. According to the resluts, the second model is better which contains less cross validaiton error. 
+
+
```

Comment:
Nice job! Very clear model comparison and selection. I appreciate that you tried to use PCA to extract/select the main features among all the variables which help to reduce the cv error rate and standard error. Maybe you could go further to explain what PC1 and CP2 consist of by looking at eigenvectors and give an initial explanation.